### ISSUE TITLE
Persist "Toggle Replace" in Project Search / Buffer Search

### ISSUE THREAD

**Issue by xcrap** [zed-industries/zed#15918]
### Describe the bug / provide steps to reproduce it

When you search for something, you click the "Toggle Replace" button that display the replace options. All good, but if you do another search it will reset the search panel and you have to click again on the "Toggle Replace" to replace again, if would be good (and I think natural) if when you click the Toggle Replace button and you keep searching this sub-panel would keep showed instead of reseting.

### Environment

Zed: v0.146.5 (Zed)
OS: macOS 14.6.0
Memory: 16 GiB
Architecture: aarch64

**Comment by 1925381584** [comment]
You can make another replace instead of another search.

**Comment by xcrap** [comment]
> You can support making another replace instead of another search.

I do know we can go a replace directly, but in terms of UI consistency I think it makes sense if you toggle something with additional opens and you don''t close the panel next time it's open it should be open with the same settings ;)

**Comment by 1925381584** [comment]
Don't think in terms of UI, think in terms of commands :)

**Comment by xcrap** [comment]
> Don't think in terms of UI, think in terms of commands :)

Of course, if you only use commands and shortcuts, it's a highly optimized and fast workflow. However, it's important to understand that some people prefer to use the UI. The UI is part of the app you use and should behave optimally. If you only use commands or shortcuts, you might as well use Vim. But since there are also UI/UX designers, the design is a crucial part of the tool.

Moreover, internationalization is vital. If you don't program in English, shortcuts are not always obvious. For example, F for "Find" and A for "All" might seem basic, but in Portuguese, it's P for "Procurar" (Find) and T for "Tudo" (All). If you don't change your keybindings, people who use other languages and multiple tools with different shortcuts for the same actions will rely more on the UI initially.

So, I believe it's all important. For me, it's all about the little details. 😊

**Comment by 1925381584** [comment]
Yeah, I thought about it some more. That interaction could really make more sense for someone thinking from the UI. Whether it's from the UI or from commands, for programming activities, thinking is about for not thinking.  😊

**Comment by xcrap** [comment]
Nothing has changes, if replaced toggle is activated in the next search is still not persistent, it was the idea of this issue.

**Comment by xcrap** [comment]
Yes, it's the same. If you toggle the replacement, it should remain active until you disable it. Right now, If you search for something, toggle the replacement, and then search again to replace something else, you have to toggle it again. This doesn't make sense to me. In my opinion, the toggle replace should be persistent and remember its last state.

**Comment by xcrap** [comment]
Okay i give up :)

**Comment by zelenenka** [comment]
@xcrap sorry about that, the bot misbehaved due to a bug (https://github.com/actions/stale/issues/1302). i've added a label that should prevent it from touching this issue again, and i will see if we can tackle this issue in the upcoming Quality Week; it doesn't look like it would be a huge fix
